### PR TITLE
Article update

### PR DESCRIPTION
**Pull request by jsuarezb**
# Summary

- Edits articles within the list of articles

# Trello card

https://trello.com/c/hOtXSVp7/22-crear-update-de-articulos-con-javascript

# Screenshots

image

### CODE REVIEW

File: app/assets/javascripts/articles.coffee [hunk 1]
```diff
@@ -1,3 +1,30 @@
 # Place all the behaviors and hooks related to the matching controller here.
 # All this logic will automatically be available in application.js.
 # You can use CoffeeScript in this file: http://coffeescript.org/
+$ ->
+  $('.edit_article').submit((e) ->
+    e.preventDefault()
+
+    url = $(this).closest('.postcnt').find('.btn-edit').data('edit-url')
+    console.log url
```
Comment (santi698) [inline app/assets/javascripts/articles.coffee]: `console.log` not allowed :scream:

File: app/assets/javascripts/articles.coffee [hunk 2]
```diff
@@ -1,3 +1,30 @@
 # Place all the behaviors and hooks related to the matching controller here.
 # All this logic will automatically be available in application.js.
 # You can use CoffeeScript in this file: http://coffeescript.org/
+$ ->
+  $('.edit_article').submit((e) ->
```
Comment (santi698) [inline app/assets/javascripts/articles.coffee]: You **may** use ids in js.

File: app/assets/javascripts/articles.coffee [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+    e.preventDefault()
+
+    url = $(this).closest('.postcnt').find('.btn-edit').data('edit-url')
+    console.log url
+    data = {}
+    params = $(this).serializeArray()
+    for x in params
+      data[x.name] = x.value
```
Comment (santi698) [inline app/assets/javascripts/articles.coffee]: Why not just use `$(this).serialize()`? You can still access data as: params['name'] in rails

File: app/assets/javascripts/articles.coffee [hunk 4]
```diff
@@ -16,6 +16,7 @@ $ ->
       method: 'PATCH'
       url: url
       data: data
+      dataType: 'json'
       success: () -> console.log('End of request')
```
Comment (santi698) [inline app/assets/javascripts/articles.coffee]: No `console.log`

File: app/assets/javascripts/articles.coffee (hunk 2 above)
Comment (jsuarezb) [inline app/assets/javascripts/articles.coffee]: There is one edit article button for each article. Using ids in this case seems fishy, right?

**Pull request merged**